### PR TITLE
Make ace work without inline styles

### PR DESCRIPTION
**Pull request by ranquild**
Related to https://github.com/metabase/metabase-private/issues/73

How to verify ace changes:
- New -> SQL query 
- The editor should be functional - without these fixes it doesn't allow you to input text

How to verify Metabase them changes:
- Admin -> Embedding -> Enable
- Question -> Share -> Embed in app -> View code
- Embedding code should be displayed and styled with our styles

Screenshot 2023-07-19 at 19 27 42

### CODE REVIEW

File: frontend/src/metabase/public/components/widgets/EmbedCodePane.jsx [hunk 1]
```diff
@@ -10,8 +10,6 @@ import {
 } from "../../lib/code";
 import CodeSample from "./CodeSample";
 
-import "metabase/lib/ace/theme-metabase";
```
Comment (ranquild) [inline frontend/src/metabase/public/components/widgets/EmbedCodePane.jsx:13]: This theme is only used here, and `TextEditor` is only used here, so I put theme styles into `TextEditor`, which is a custom wrapper for ace used here.

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ import {
 } from "./ExpressionEditorTextfield.styled";
 
 ace.config.set("basePath", "/assets/ui/");
+ace.config.set("useStrictCSP", true);
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx:29]: From their demo here https://github.com/ajaxorg/ace-builds/blob/master/demo/csp.html

File: frontend/src/metabase/components/TextEditor/TextEditor.jsx [hunk 3]
```diff
@@ -38,7 +40,7 @@ export default class TextEditor extends Component {
   }
 
   _update() {
-    const element = ReactDOM.findDOMNode(this);
```
Comment (ranquild) [inline frontend/src/metabase/components/TextEditor/TextEditor.jsx:41]: The change isn't required but was an easy fix. Minus 1 warning in the console.

**Review by kamilmielnik — APPROVED** [review]
Tested, looks good :+1:

**Pull request merged**